libraryAPI/pkg/repository/repository.go

```diff
@@ -0,0 +1,20 @@
+package repository
+
+import (
+	"library/pkg/model"
+)
+
+type Book interface {
+	CreateBooks(users []model.Book) (int, error)
+	GetBooks() ([]model.Book, error)
+}
+
+type Repository struct {
+	Book
+}
+
+func NewRepository() *Repository {
+	return &Repository{
+		Book: NewBookRepository(),
+	}
+}
```

Comment:
i would have split repository.go and book_memory.go
move the Book interface to book_memory.go file and in 'NewBookRepository' func return Book interface instead of BookRepository struct's object, i would even rename interface to BookRepository and rename struct as bookRepository